sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/IdGenerator.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2023 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.sdk.trace
+
+import cats.Monad
+import cats.effect.std.Random
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.monad._
+import org.typelevel.otel4s.trace.SpanContext
+import scodec.bits.ByteVector
+
+/** Generates new span and trace ids.
+  *
+  * @see
+  *   [[https://opentelemetry.io/docs/specs/otel/trace/sdk/#id-generators]]
+  *
+  * @tparam F
+  *   the higher-kinded type of a polymorphic effect
+  */
+trait IdGenerator[F[_]] {
```

Comment:
I hope we don't get screwed over by compatibility 😇 probably not, but the `F[_]` makes it impossible for us to add default implementations of new methods where other languages could.